barcode_validation/models/barcode_validation_mixin.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class BarcodeValidation(models.AbstractModel):
+    _name = "barcode.validation.mixin"
+    _description = "Barcode Validation"
+
```

Comment:
suggestion: I don't know if the answer to https://github.com/OCA/stock-logistics-barcode/pull/607#discussion_r1591886581 will imply that you don't need a mixin anymore. But in case you still need it, I think the mixin is incomplete. I'd add this:

```suggestion
    _barcode_field = "barcode"  # Override if needed when applying mixin
    
    @api.constrains(lambda self: (self._barcode_field,))
    def _check_barcode(self):
        for record in self:
            record._validate_barcode(record[self._barcode_field])

    @api.model
```